lib/mage/MageError.ts

```diff
@@ -0,0 +1,53 @@
+/**
+ * MageError details
+ */
+declare interface IErrorDetails {
+    /**
+     * The log level to use when logging this error (default: error)
+     *
+     * Some types of errors are perfectly normal; for instance,
+     * a user failing to authenticate may throw an error, but
+     * you might not want to have it logged as an error.
+     *
+     * Setting the level will allow you to decide whether this
+     * error should be ignored, considered benign, or serious.
+     */
+    level?: 'debug' | 'warning' | 'error' | 'alert' | 'emergency',
+
+    /**
+     * The error code
+     *
+     * This code will generally be the code that will be returned
+     * to the client if this error is thrown during a user command
+     * request.
+     */
+    code: string,
```

Comment:
I believe `state` allows `code` to be both `string` or `number`.

Reply:
What happens if its a number?

Reply:
I see the code is internally stringified.

Personally, I would prefer to be explicit at least in typescript about this - I'd hate to see confusion with a dev returning a number only to wonder why it's being received as a string on the client.